src/core/store.ts

```diff
@@ -0,0 +1,653 @@
+            if (canUnmountAtom(a, mounted)) {
+              unmountAtom(a)
+            }
+          }
+        }
+      })
+    } else if (
+      typeof process === 'object' &&
+      process.env.NODE_ENV !== 'production'
+    ) {
+      console.warn('[Bug] could not find atom state to unmount', atom)
+    }
+  }
+
+  const mountDependencies = <Value>(
+    atom: Atom<Value>,
+    atomState: AtomState<Value>,
+    prevDependencies: ReadDependencies
+  ): void => {
+    const dependencies = new Set(atomState.d.keys())
+    prevDependencies.forEach((_, a) => {
+      if (dependencies.has(a)) {
+        // not changed
+        dependencies.delete(a)
+        return
+      }
+      const mounted = mountedMap.get(a)
+      if (mounted) {
+        mounted.d.delete(atom)
+        if (canUnmountAtom(a, mounted)) {
+          unmountAtom(a)
+        }
+      }
+    })
+    dependencies.forEach((a) => {
+      const mounted = mountedMap.get(a)
+      if (mounted) {
+        const dependents = mounted.d
+        dependents.add(atom)
+      } else {
+        mountAtom(a, atom)
+      }
+    })
+  }
+
+  const commitAtomState = <Value>(
+    atom: Atom<Value>,
+    atomState: AtomState<Value>,
+    prevDependencies?: ReadDependencies
+  ): void => {
+    if (typeof process === 'object' && process.env.NODE_ENV !== 'production') {
+      Object.freeze(atomState)
+    }
+    const isNewAtom = !atomStateMap.has(atom)
+    atomStateMap.set(atom, atomState)
+    if (stateListener) {
+      stateListener(atom, isNewAtom)
+    }
+    ++version
+    if (!pendingMap.has(atom)) {
+      pendingMap.set(atom, prevDependencies)
+    }
+  }
+
+  const flushPending = (): void => {
+    const pending = Array.from(pendingMap)
+    pendingMap.clear()
+    pending.forEach(([atom, prevDependencies]) => {
+      const atomState = getAtomState(atom)
+      if (atomState) {
+        if (prevDependencies) {
+          mountDependencies(atom, atomState, prevDependencies)
+        }
+      } else if (
+        typeof process === 'object' &&
+        process.env.NODE_ENV !== 'production'
+      ) {
+        console.warn('[Bug] atom state not found in flush', atom)
+      }
+      const mounted = mountedMap.get(atom)
+      mounted?.l.forEach((listener) => listener())
+    })
+  }
+
+  const subscribeAtom = (atom: AnyAtom, callback: () => void) => {
+    const mounted = addAtom(atom)
+    const listeners = mounted.l
+    listeners.add(callback)
+    return () => {
+      listeners.delete(callback)
+      delAtom(atom)
+    }
+  }
+
+  const restoreAtoms = (
+    values: Iterable<readonly [AnyAtom, unknown]>
+  ): void => {
+    for (const [atom, value] of values) {
+      if (hasInitialValue(atom)) {
+        setAtomValue(atom, value)
+        invalidateDependents(atom)
+      }
+    }
+    flushPending()
+  }
+
+  if (typeof process === 'object' && process.env.NODE_ENV !== 'production') {
+    return {
+      [GET_VERSION]: () => version,
+      [READ_ATOM]: readAtom,
```

Comment:
What's the reasoning behind this architecture? It doesn't help readability.

Reply:
This allows to minify code. The same reason why previously it was function style. I hope it's more readable than `r: readAtom`.